client/components/backup-retention-management/hooks.tsx

```diff
@@ -0,0 +1,77 @@
+import { useMemo } from '@wordpress/element';
+import { useSelector } from 'react-redux';
+import { useStorageText } from 'calypso/components/backup-storage-space/hooks';
+import getBackupCurrentSiteSize from 'calypso/state/rewind/selectors/get-backup-current-site-size';
+import getRewindBytesAvailable from 'calypso/state/rewind/selectors/get-rewind-bytes-available';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { STORAGE_ESTIMATION_ADDITIONAL_BUFFER } from './constants';
+import type { RetentionRadioOptionType } from './types';
+
+/**
+ * Estimate the current site size based on the last backup size, adding a 25% buffer
+ */
+export function useEstimatedCurrentSiteSize(): number {
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const lastBackupSize = useSelector(
+		( state ) => getBackupCurrentSiteSize( state, siteId ) as number
+	);
+	return useMemo(
+		() => lastBackupSize + lastBackupSize * STORAGE_ESTIMATION_ADDITIONAL_BUFFER,
+		[ lastBackupSize ]
+	);
+}
+
+/**
+ * Estimate the space needed based on the current site size and the retention days
+ *
+ * @param currentSiteSizeInBytes Current site size in bytes
+ * @param desiredRetentionDays   Desired backup retention days
+ * @returns Space needed in bytes
+ */
+export function useEstimateSpaceNeeded(
+	currentSiteSizeInBytes: number,
+	desiredRetentionDays: number
+): number {
+	return useMemo( () => {
+		return currentSiteSizeInBytes * desiredRetentionDays;
+	}, [ currentSiteSizeInBytes, desiredRetentionDays ] );
```

Comment:
Have we evaluated the need for each of those memoizations? What's the performance impact and why do we need them? Can we do without them? Would be good to measure and use that to decide whether we need them.

Reply:
We did some refactors on the way we handle the retention options, and we needed to stop using hooks, as they must only be called at top level.

Either way, at first sight and given that they are simple math operations and the fact that we are not calling too many times those functions, the memoization is not required for `useEstimatedCurrentSiteSize()` and `useEstimateSpaceNeeded()`.

Addressed in #73303